src/lib/logstore/log_dev.cpp

```diff
@@ -255,31 +250,33 @@ void LogDev::assert_next_pages(log_stream_reader& lstream) {
     }
 }
 
-int64_t LogDev::append_async(const logstore_id_t store_id, const logstore_seq_num_t seq_num, const sisl::io_blob& data,
-                             void* cb_context, bool flush_wait) {
-    auto prev_size = m_pending_flush_size.fetch_add(data.size(), std::memory_order_relaxed);
+int64_t LogDev::append_async(logstore_id_t store_id, logstore_seq_num_t seq_num, const sisl::io_blob& data,
+                             void* cb_context) {
     const auto idx = m_log_idx.fetch_add(1, std::memory_order_acq_rel);
-    auto threshold_size = LogDev::flush_data_threshold_size();
     m_log_records->create(idx, store_id, seq_num, data, cb_context);
+    m_pending_flush_size.fetch_add(data.size(), std::memory_order_relaxed);
+    if (allow_inline_flush()) flush_if_necessary();
+    return idx;
+}
 
-    if (HS_DYNAMIC_CONFIG(logstore.flush_threshold_size) == 0) {
-        // This is set in tests to disable implicit flush. This will be removed in future.
-        return idx;
+log_buffer LogDev::read_from_cache(const logid_t& idx) {
+    sisl::io_blob buf;
+    try {
+        buf = m_log_records->at(idx).data;
+    } catch (const std::out_of_range& e) {
```

Comment:
ok, let`s do it follow our decision, so that we do not make any optimization about home_log_store, it will work like before.

we only make the optimization at home_raft_log_store level. 

@xiaoxichen 
for home_log_store, it will always flush and read from vdev like before, so there will not be any potential issue about the safe buffer